core/variant/variant_construct.cpp

```diff
@@ -303,7 +303,7 @@ void VariantInternal::object_assign(Variant *v, const Object *o) {
 	if (o) {
 		if (o->is_ref_counted()) {
 			RefCounted *ref_counted = const_cast<RefCounted *>(static_cast<const RefCounted *>(o));
-			if (!ref_counted->init_ref()) {
+			if (ref_counted->reference_get_count() == 0) {
```

Comment:
For the record, this looks like it was copied/adapted from the `Variant` constructor:
https://github.com/godotengine/godot/blob/a56e8f8afc2b358ec6e58eed2725bfe85942e5e6/core/variant/variant.cpp#L2516-L2531

I'm not really familiar with this code, but won't this mean that when you assign a RefCounted object to a variant, it won't increment the refcount? So if it's dropped in whichever other object holds a reference to it, wouldn't this cause issues?

Reply:
Whoops. This needs to be its own function for this specific use case. I didn't notice it had other usage.

OPCODE_CALL_PTRCALL_OBJECT handles the return value of a native method returning an Object. When this object is a Ref<>, the native call automatically increments the refcount on return. So, object_assign redundantly increments the refcount in this case.